### PR TITLE
Feature: logs view

### PR DESCRIPTION
**Pull request by ricklamers**
## Description

This PR adds a full screen log viewer to the pipeline view. It depends on https://github.com/orchest/orchest/pull/224 and should be merged after that lands on master.

The log view allows you to inspect the logs of any pipeline step and service.

### Checklist
- [ ] The documentation reflects the changes.
- [ ] I have manually tested the application to make sure the changes don’t cause any downstream issues, which includes making sure `./orchest status --ext` is not reporting failures when Orchest is running.

### CODE REVIEW

**Comment by yannickperrenet** [comment]
>  It depends on #224 and should be merged after that lands on master.

We could simply merge this branch in the `feature-services` branch first and then merge `feature-services` into `master`.

File: services/orchest-webserver/client/src/components/LogViewer.jsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
+import React from "react";
+import { RefManager, uuidv4 } from "@orchest/lib-utils";
+import { XTerm } from "xterm-for-react";
+import { FitAddon } from "xterm-addon-fit";
+
+class LogViewer extends React.Component {
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/components/LogViewer.jsx:6]: All new components in `orchest-webserver` should be in the Functional Component syntax please 🙏 
(See #266)

File: services/orchest-webserver/client/src/views/LogsView.jsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import io from "socket.io-client";
+import {
+  MDCButtonReact,
+  MDCLinearProgressReact,
+  MDCDrawerReact,
+} from "@orchest/lib-mdc";
+import { OrchestContext, OrchestSessionsConsumer } from "@/hooks/orchest";
+import { getPipelineJSONEndpoint } from "../utils/webserver-utils";
+
+class LogsView extends React.Component {
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/views/LogsView.jsx:19]: Similarly to above, please convert this to a Functional Component

**Review by joe-bell — CHANGES_REQUESTED** [review]
Generally happy with the new front-end changes!

My only request is that we make sure any new components/views within `orchest-webserver` are written as typed Functional Components (#259) with `// @ts-check`, otherwise we’re just needlessly adding to the pile of components that need to be converted. This'll enable us to make crucial improvements to context/shared state in the future

There are plenty of existing examples in the codebase that should help, but I’m also happy to pair on it if you get stuck

File: services/orchest-webserver/client/src/components/LogViewer.jsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
+import React from "react";
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/components/LogViewer.jsx:1]: Don't forget type checks!

```suggestion
// @ts-check
import React from "react";
```

File: services/orchest-webserver/client/src/views/LogsView.jsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
+import React from "react";
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/views/LogsView.jsx:1]:
```suggestion
// @ts-check
import React from "react";
```

File: services/orchest-webserver/client/src/components/LogViewer.jsx (hunk 1 above)
Comment (ricklamers) [reply services/orchest-webserver/client/src/components/LogViewer.jsx:6]: This is actually a component rename, so I'll leave this a class component for now. It's relatively complex with the socket.io connection handling, so I'd rather do that refactor when we have more time.

File: services/orchest-webserver/client/src/views/LogsView.jsx (hunk 2 above)
Comment (ricklamers) [reply services/orchest-webserver/client/src/views/LogsView.jsx]: Good idea! Refactored this to a functional component.

File: services/orchest-webserver/client/src/components/LogViewer.jsx (hunk 3 above)
Comment (ricklamers) [reply services/orchest-webserver/client/src/components/LogViewer.jsx:1]: Good catch.

File: services/orchest-webserver/client/src/views/LogsView.jsx (hunk 4 above)
Comment (ricklamers) [reply services/orchest-webserver/client/src/views/LogsView.jsx:1]: Good catch!

**Comment by ricklamers** [comment]
Moving this PR to a new PR that merges into feature-services.

**Comment by ricklamers** [comment]
> Now that logs have their own button in the pipeline editor, should they be removed from the properties pane?

I think they can stay there, the log viewing component is being re-used. Checking the logs there is less invasive (pane instead full view), so I like it as an addition.

> When opening the logs view I find it hard to navigate to the step I want to see the logs for. What about topological sorting? Would love to hear other ideas ;)

Like the idea! Implemented a topological sort that made sense to me.

**Pull request closed without merging**